`brew install io --with-python` broken on 10.13

# Please always follow these steps:
- [X] Confirmed this is a problem with `brew install`ing one, specific Homebrew/homebrew-core formula (not cask or tap) and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Homebrew/brew/issues/new. If it's a `brew cask` problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [X] Ran `brew update` and retried your prior step?
- [X] Ran `brew doctor`, fixed all issues and retried your prior step?
- [X] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

------

`brew install io --with-python` is failing for me with a failure to find header file.

```
In file included from /tmp/io-20180211-76767-1e0phii/io-2017.09.06/addons/Python/source/PythonData.c:3:
/tmp/io-20180211-76767-1e0phii/io-2017.09.06/addons/Python/source/PythonData.h:4:10: fatal error: 'Python.h' file not found
#include <Python.h>
         ^~~~~~~~~~
1 error generated.
make[2]: *** [addons/Python/CMakeFiles/IoPython.dir/source/PythonData.c.o] Error 1
make[1]: *** [addons/Python/CMakeFiles/IoPython.dir/all] Error 2
make: *** [all] Error 2
```

Build logs: https://gist.github.com/4fa836138fed5798682e9617222d165b

Looks like maybe it's trying to build against Python 3, while our dependency is for python 2?

```

clang called with: -DIoPython_EXPORTS -I/tmp/io-20180211-76767-1e0phii/io-2017.09.06/libs/coroutine/source -I/tmp/io-20180211-76767-1e0phii/io-2017.09.06/libs/basekit/source -I/tmp/io-20180211-76767-1e0phii/io-2017.09.06/libs/garbagecollector/source -I/tmp/io-20180211-76767-1e0phii/io-2017.09.06/libs/iovm/source -I/usr/local/Frameworks/Python.framework/Versions/3.6/include/python3.6m -fobjc-exceptions -DNDEBUG -fPIC -o CMakeFiles/IoPython.dir/source/PythonData.c.o -c /tmp/io-20180211-76767-1e0phii/io-2017.09.06/addons/Python/source/PythonData.c
superenv removed:  -I/usr/local/Frameworks/Python.framework/Versions/3.6/include/python3.6m
```

--- a/Formula/io.rb
+++ b/Formula/io.rb
@@ -33,7 +33,7 @@ class Io < Formula
     depends_on "pcre"
     depends_on "yajl"
     depends_on "xz"
-    depends_on "python" => :optional
+    depends_on "python3" => :optional
   end
 
   def install
@@ -48,7 +48,7 @@ class Io < Formula
                                   "#add_subdirectory(addons)"
     else
       inreplace "addons/CMakeLists.txt" do |s|
-        if build.without? "python"
+        if build.without? "python3"
           s.gsub! "add_subdirectory(Python)", "#add_subdirectory(Python)"
         end
 

